rrabinovitch/Crate
# User queries testing

#### What’s this PR do?
tests `user` `getAll` and `getById` queries. beyond was what covered in class today, I also added expectations re: style profile and email

#### Where should the reviewer start?
look at `src/modules/user/query.test.js` for test content

#### How should this be manually tested?
Once you've completed the setup below, make sure you're in the `code/api` directory and run `npm test` - see screenshot of what output should look like below. If the tests don't run immediately, run `a` to run all tests and confirm they are passing. And then `q` to quit the test interface.

#### Any background context you want to provide?
This PR adds Jest and SuperTest packages to the `code/api` directory. I'm not sure if they need to be installed locally as well, so if they do the process was as follows (assuming you're starting from the root directory):
_installing the packages..._
`cd code/api`
`npm install --save-dev jest`
`npm install supertest --save-dev`
_creating test database (if you haven't already)..._
`psql`
`CREATE DATABASE crate_test;`
`exit`
`NODE_ENV=test npm run setup:db`

**This PR does not include test teardown.** I did some research on it and included the code that I think will address teardown, but since I have not yet had any errors, I commented the code out so we can run into an error first and know for sure whether this teardown snippet actually addresses the error and is doing what it's expected to do (or keep troubleshooting if it doesn't address the error).

#### What are the relevant tickets?
closes #32 

#### Screenshots (if appropriate)
<img width="650" alt="Screen Shot 2020-10-13 at 12 53 11 PM" src="https://user-images.githubusercontent.com/62635544/95903187-0dce4280-0d53-11eb-90d7-1be2556afb2f.png">


#### Questions:  
- hoping I rebased correctly...?


--------
- [X] I checked that the tests are running successfully before submitting this PR
